### ISSUE TITLE
Change of pytest_xdist default breaks (at least) calibre

### ISSUE THREAD

**Issue by mdorman** [NixOS/nixpkgs#106058]
**Describe the bug**

installing 'calibre-4.23.0'
error: pytest-xdist-2.1.0 not supported for interpreter python2.7
(use '--show-trace' to show detailed location information)

**To Reproduce**

Attempt to instantiate `calibre-py2`

**Expected behavior**

No error.

**Additional context**

A consequence of e0cbbc6f30350ad825350ec1177cf104b05f4c84

**Notify maintainers**

@jonringer created the change, @FRidh commited it

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.11, NixOS, 21.03.git.1241d335d1e (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `""`
 - channels(mdorman): `""`
 - nixpkgs: `/var/nixup/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by TredwellGit** [comment]
https://github.com/NixOS/nixpkgs/pull/105878

**Comment by FRidh** [comment]
Fixed with #105878
